packages/lightning-lsp-common/src/shared.ts

```diff
@@ -73,25 +73,44 @@ export function detectWorkspaceHelper(root: string): WorkspaceType {
         return WorkspaceType.CORE_PARTIAL;
     }
 
+    if (fs.existsSync(path.join(root, 'lwc.config.js'))) {
+        return WorkspaceType.STANDARD_LWC;
+    }
+
     const packageJson = path.join(root, 'package.json');
     if (fs.existsSync(packageJson)) {
         try {
-            // Check if package.json contains @lwc/engine
             const packageInfo = JSON.parse(fs.readFileSync(packageJson, 'utf-8'));
             const dependencies = Object.keys(packageInfo.dependencies || {});
-            if (dependencies.includes('@lwc/engine')) {
+            const devDependencies = Object.keys(packageInfo.devDependencies || {});
+            const allDependencies = [...dependencies, ...devDependencies];
+            const hasLWCdependencies = allDependencies.some(key => key.includes('@lwc/'));
+
+            // any type of @lwc is a dependency
+            if (hasLWCdependencies) {
                 return WorkspaceType.STANDARD_LWC;
             }
-            const devDependencies = Object.keys(packageInfo.devDependencies || {});
-            if (devDependencies.includes('@lwc/engine')) {
+
+            // has and type of lwc configuration
+            if (packageInfo.lwc) {
                 return WorkspaceType.STANDARD_LWC;
             }
+
+            // has a lwc-services dependency (this ca be removed once the
+            // `lwc.config.js` becomes required
+            if (allDependencies.includes('lwc-services')) {
```

Comment:
Are we considering to merge "Add Support for LWC OSS" PR #16 by @muenzpraeger ? Looks like that PR does some more language support scenarios specific to apps created by `create-lwc-app`. It also checks `lwc-services` [here](https://github.com/forcedotcom/lightning-language-server/pull/16/files#diff-b7cf3b6b041b8f0e826428e4615fbe5fR92-R94).

@muenzpraeger I also wonder if we are going to make a distinction in the future between:
- Generic LWC OSS projects

Quoting @mysticflute: (The latter ones)
> Some OSS users will use create-lwc-app , and apps created for demos, but internal off-core and core projects won't be using it. They will either have a dependency on @lwc, or have an lwc.config.json or have lwc config in the package.json
> Here's an example of an off-core project that is using the lwc config https://github.com/salesforce/lightning-components/blob/master/ui-lightning-components/package.json#L111
- Apps created by `create-lwc-app`

For now, it would help with extension capabilities (like determining a proper test runner for `create-lwc-app` apps)

Reply:
I think @lcampos had mentioned the PR #16 needs some rework, maybe we can leave this be now.

Reply:
There are a couple of things, also based on a convo with @mysticflute:

- I will add support to create-lwc-app to use `lwc.config.js` - so we won’t need the additional check
- The module resolution mechanism, and the shape for the config of `lwc.config.js` has a breaking change after/with 1.4.0.

Now, coming back to no 1, I think we can remove the lwc-services check.

For no 2 there needs to be a decision. Latest with 228 LWC 1.4.0 will be internally also usable/used. In OSS (and likely many off core projects it’s also used). I am happy to put in a new PR, based on this PR, who makes usage of the new module resolution. 

One caveat, in general, is that the LWC versions for on and off platform  deviate. I would propose to update to 1.5.x. for the tooling here in general.

Thoughts @lcampos @diervo ?

Reply:
We have a story to update the LWC version in this project but unfortunately it will be some work because a lot of the types have been removed to the platform compiler or removed.